File: Backend/Game Buddy Finder/Game Buddy Finder/Models/Comment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Game_Buddy_Finder.Models
{
    public class Comment
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int CommentId { get; set; }

        [Required]
        public int PostId { get; set; }

        [Required]
        public int PosterUserId { get; set; }

        [Required]
        public string Content { get; set; }

        [Required]
        public DateTime PostTime { get; set; }

        [NotMapped]
        public User PosterUser { get; set; }

        [NotMapped]
        public Profile PosterProfile { get; set; }
    }
}

File: Frontend/game-buddy-finder/src/app/models/comment.ts
import { User } from './user'
export class Comment {
    commentId: number;          //Server Generated
    postId: number;
    posterUserId: number;
    content: string;
    postTime: VarDate;          //Server Generated
    Poster: User;               //Server Generated

    public constructor(postId: number, posterUserId: number, content: string) {
        this.postId = postId;
        this.posterUserId = posterUserId;
        this.content = content;
    }
}